src/engine/contactsengine.cpp

```diff
@@ -822,23 +870,31 @@ QString ContactsEngine::databaseUuid()
 
 QContactManager::Error ContactsEngine::open()
 {
-    if (m_database.open(QString(QLatin1String("qtcontacts-sqlite-%1")).arg(databaseUuid()), m_nonprivileged)) {
-        setNonprivileged(m_database.nonprivileged());
-
-        m_notifier.reset(new ContactNotifier(m_nonprivileged));
-        m_notifier->connect("contactsAdded", "au", this, SLOT(_q_contactsAdded(QVector<quint32>)));
-        m_notifier->connect("contactsChanged", "au", this, SLOT(_q_contactsChanged(QVector<quint32>)));
-        m_notifier->connect("contactsPresenceChanged", "au", this, SLOT(_q_contactsPresenceChanged(QVector<quint32>)));
-        m_notifier->connect("syncContactsChanged", "as", this, SLOT(_q_syncContactsChanged(QStringList)));
-        m_notifier->connect("contactsRemoved", "au", this, SLOT(_q_contactsRemoved(QVector<quint32>)));
-        m_notifier->connect("selfContactIdChanged", "uu", this, SLOT(_q_selfContactIdChanged(quint32,quint32)));
-        m_notifier->connect("relationshipsAdded", "au", this, SLOT(_q_relationshipsAdded(QVector<quint32>)));
-        m_notifier->connect("relationshipsRemoved", "au", this, SLOT(_q_relationshipsRemoved(QVector<quint32>)));
-        return QContactManager::NoError;
-    } else {
-        QTCONTACTS_SQLITE_WARNING(QString::fromLatin1("Unable to open engine database"));
-        return QContactManager::UnspecifiedError;
+    // Start the async thread, and wait to see if it can open the database
+    if (!m_jobThread) {
+        m_jobThread.reset(new JobThread(this, databaseUuid(), m_nonprivileged, m_autoTest));
+
+        if (m_jobThread->databaseOpen()) {
+            // We may not have got privileged access if we requested it
+            setNonprivileged(m_jobThread->nonprivileged());
+
+            if (!m_notifier) {
```

Comment:
No, it can't be reused, this is only entered when the jobThread is first initialized.  By the time we have opened the database, we know whether we have the privileged or NP database.